### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by superboy8900** [rayarachelian/lisaem#12]
When I start to build lisaem it give me an error saying
`home/jandaseamansfamily/lisaem/waitq: Aborted due to a failed job! See /home/jandaseamansfamily/lisaem/obj/build-warnings.txt`

I am using debian linux with wxWidgets 3.0.5

**Comment by rayarachelian** [comment]
Right, so that means there was a compilation failure. The actual error message will be either above that line of text, or if not, open up the file /home/jandaseamansfamily/lisaem/obj/build-warnings.txt in a text editor, and then look at the bottom of that file for the actual error.

The text:
```home/jandaseamansfamily/lisaem/waitq: Aborted due to a failed job! See /home/jandaseamansfamily/lisaem/obj/build-warnings.txt```
does not tell me enough to be able to help you debug this error.

That said, I hope you're not using the wxWidgets that came with debian linux, and instead have compiled something like wxWidgets-3.1.5 using the script inside the scripts directory ( [build-wx3.1.5-gtk.sh](https://github.com/rayarachelian/lisaem/blob/master/scripts/build-wx3.1.5-gtk.sh) ), and then have set your PATH variable to point to it before you actually compile Lisaem itself. While those system wxWdigets packages may work, they'll usually have issues.

**Comment by superboy8900** [comment]
Here is that file

[build-warnings.txt](https://github.com/rayarachelian/lisaem/files/9329468/build-warnings.txt)

**Comment by superboy8900** [comment]
Also I updated to wxWidgets 3.1.5 and that did not fix it

**Comment by rayarachelian** [comment]
So, according to the linker error in build-warnings.txt, you're clearly not actually using a wxWidgets that was built from the scripts directory. 

Those paths are not supposed to look like:  /usr/local/lib/libwx*, but rather should look like `/usr/local/wx3.1.5-gtk` and if you run which wx-config, it should show up as `/usr/local/wx3.1.5-gtk/bin/wx-config` and not something like `/usr/bin/wxwidgets`.

Assuming you've built wxWidgets from the script, but haven't updated the path, this would help:

```
export PATH=/usr/local/wx3.1.5-gtk/bin/:$PATH
```

Which you can verify by running `which wx-config` and it should say `/usr/local/wx3.1.5-gtk/bin/wx-config` before you run `./build.sh clean build` from the lisaem folder.

**Comment by superboy8900** [comment]
Another error

[build-warnings.txt](https://github.com/rayarachelian/lisaem/files/9329632/build-warnings.txt)

**Comment by rayarachelian** [comment]
I see the correct path for wxWidgets now, but not sure what's stopping the compilation, there's no actual error message in the log. Is there anything on the display that indicates where it stopped?

It looks like it built libGenerator and then libdc42 and then the tools, but I don't see it trying to compile Lisaem itself.

**Comment by rayarachelian** [comment]
Can you tell me what version of Debian you're using? Maybe I can spin it up as a VM and try to see where it's failing.
Also what version of lisaem are you using? master or unstable branch, or a specific version?

**Comment by superboy8900** [comment]
also it keeps changing the path of wxWidgets

**Comment by superboy8900** [comment]
I am using Bullseye and using the most recent lisaem that you made

**Comment by rayarachelian** [comment]
So I've installed bullseye in virtualbox with 4MB RAM + 8GB virtual hard drive, and just GNOME desktop, and then allowed it to reboot the VM.
I installed gcc-10, and git.
I then did the following as per: https://www.linuxbabe.com/desktop-linux/how-to-install-virtualbox-guest-additions-on-debian-step-by-step

```
apt-get update && apt-get upgrade
apt-get install build-essential module-assistant
m-a prepare
```
 * I ran the virtualbox addition for Linux from the Guest Additions CD and rebooted with `init 6`
 * I reset the display resolution higher than 800x600.
 * I launched Terminal and did `sudo -s`
 * I then went on to install the GTK developer libs as required by wxWidgets `apt install libgtk-3-dev` and then `apt install libtiff-dev`
 * I did `git clone https://github.com/rayarachelian/lisaem`
 * Next, I ran `cd lisaem/scripts` and `./build-wx3.1.5-gtk.sh` which took quite a while as expected.
 * I copy-pasta'ed the export PATH statement and confirmed `which wx-config` was correct.
 * I then did `cd ..` and then `./build.sh`
 * I saw the build.sh script build libGenerator, then libdc42, then the tools, then it started to build the code for LisaEm
 * It then built the TerminalWx library, then the C files for LisaEm, and then finally, the C++ portions of LisaEm.
 * and that produced a working LisaEm binary.

Now, I cheated a bit by using the root account in Terminal to do all this, not sure if that's the differential.

Finally, I did `./build.sh install` as I was logged in as user `ray` and XWayland wouldn't allow me to run the lisaem binary, as the desktop was not logged in as root, but then switching to another Terminal window, I was then able to run lisaem and bring it up (since user ray can't access root's home directory, doing `./build.sh install` allowed it to launch.)

Sadly, I wasn't able to reproduce the build failure, but perhaps you could try first doing `sudo -s` and then setting the PATH again, and doing `./build.sh clean build install` and see if that helps anything.

**Comment by superboy8900** [comment]
It works!! I also forgot to tell you that I am using Crostini a linux emulator on a chromebook that runs debian bullseye. But still now It works because of your help. Thank you so much

**Comment by rayarachelian** [comment]
Sure, no problem, if you had said it didn't work, I would have done `./build.sh package` and give you the .deb, but if your chromebook is not on Intel, it might not have worked.

I've got a couple of Acer R11 Intel based chromebooks that I installed GalliumOS on, didn't use Crostini, but, good enough. :) Sadly, GalliumOS is not getting new updates, oh well.

Were you able to figure out what was different? Might be useful for me to add to the documentation.
